File: iosApp/iosApp/Features/Profile/components/HelpAndSupport.swift
//
//  HelpAndSupport.swift
//  iosApp
//
//  Created by velkonost on 20.09.2023.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import SharedSDK

struct HelpAndSupport: View {
    
    let onContactUsClick: () -> Void
    let onTelegramClick: () -> Void
    
    init(onContactUsClick: @escaping () -> Void, onTelegramClick: @escaping () -> Void) {
        self.onContactUsClick = onContactUsClick
        self.onTelegramClick = onTelegramClick
    }
    
    var body: some View {
        SectionTitle(text: SharedR.strings().profile_help_title.desc().localized())
        
        PrimaryBox {
            VStack {
                MenuItem(
                    title: SharedR.strings().profile_help_contact_support.desc().localized(),
                    icon: SharedR.images().ic_chat_cloud.toUIImage()!,
                    onClick: onContactUsClick
                )
                
                MenuItem(
                    title: SharedR.strings().tg_group_title.desc().localized(),
                    icon: SharedR.images().ic_telegram.toUIImage()!,
                    onClick: onTelegramClick
                )
            }
        }
    }
}

File: androidApp/features/profile/src/main/kotlin/com/velkonost/getbetter/android/features/profile/components/HelpAndSupport.kt
package com.velkonost.getbetter.android.features.profile.components

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.velkonost.getbetter.core.compose.components.PrimaryBox
import com.velkonost.getbetter.shared.resources.SharedR
import dev.icerock.moko.resources.compose.painterResource
import dev.icerock.moko.resources.compose.stringResource

@Composable
fun HelpAndSupport(
    modifier: Modifier = Modifier,
    onContactUsClick: () -> Unit,
    onTelegramClick: () -> Unit
) {
    SectionTitle(text = stringResource(resource = SharedR.strings.profile_help_title))

    PrimaryBox {
        Column {
//            MenuItem(
//                title = stringResource(resource = SharedR.strings.profile_help_rate),
//                icon = painterResource(imageResource = SharedR.images.ic_empty_star),
//            ) {
//
//            }
            MenuItem(
//                modifier = modifier.padding(top = 4.dp),
                title = stringResource(resource = SharedR.strings.profile_help_contact_support),
                icon = painterResource(imageResource = SharedR.images.ic_chat_cloud),
                onClick = onContactUsClick
            )

            MenuItem(
                modifier = modifier.padding(top = 4.dp),
                title = stringResource(resource = SharedR.strings.tg_group_title),
                icon = painterResource(imageResource = SharedR.images.ic_telegram),
                onClick = onTelegramClick
            )
        }
    }
}